File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/OfflineApi.swift
import Foundation

/// Facade for offline verification and bundle management.
public struct OfflineApi {
    let offlineVerifier: OfflineVerifier
    let bundleManager: BundleManager
    let orchestrator: VerificationOrchestrator

    public func prefetchBundle(issuerDid: String, statusListUrl: String? = nil) async -> Result<VerificationBundle, Error> {
        await bundleManager.prefetchBundle(issuerDid: issuerDid, statusListUrl: statusListUrl)
    }

    public func refreshStaleBundles() async -> Int {
        await bundleManager.refreshStaleBundles()
    }

    public func hasFreshBundle(issuerDid: String) async -> Bool {
        await bundleManager.hasFreshBundle(issuerDid: issuerDid)
    }

    public func verifyOffline(_ credential: VerifiableCredential) async -> OfflineVerificationResult {
        await offlineVerifier.verifyCredential(credential)
    }

    public func verify(_ credential: VerifiableCredential) async -> UnifiedVerificationResult {
        await orchestrator.verify(credential: credential)
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/OfflineApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.VerifiableCredential
import my.ssdid.sdk.domain.verifier.offline.BundleManager
import my.ssdid.sdk.domain.verifier.offline.OfflineVerificationResult
import my.ssdid.sdk.domain.verifier.offline.OfflineVerifier
import my.ssdid.sdk.domain.verifier.offline.UnifiedVerificationResult
import my.ssdid.sdk.domain.verifier.offline.VerificationBundle
import my.ssdid.sdk.domain.verifier.offline.VerificationOrchestrator

class OfflineApi internal constructor(
    private val offlineVerifier: OfflineVerifier,
    private val bundleManager: BundleManager,
    private val orchestrator: VerificationOrchestrator
) {
    suspend fun prefetchBundle(issuerDid: String, statusListUrl: String? = null): Result<VerificationBundle> =
        bundleManager.prefetchBundle(issuerDid, statusListUrl)
    suspend fun refreshStaleBundles(): Int = bundleManager.refreshStaleBundles()
    suspend fun hasFreshBundle(issuerDid: String): Boolean = bundleManager.hasFreshBundle(issuerDid)
    suspend fun verifyOffline(credential: VerifiableCredential): OfflineVerificationResult =
        offlineVerifier.verifyCredential(credential)
    suspend fun verify(credential: VerifiableCredential): UnifiedVerificationResult =
        orchestrator.verify(credential)
}
